composer/utils/object_store/mlflow_object_store.py

```diff
@@ -0,0 +1,371 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""MLFlow Artifacts object store."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import uuid
+from typing import Callable, List, Optional, Tuple, Union
+
+import mlflow
+from databricks.sdk import WorkspaceClient
+from mlflow import MlflowClient
+from mlflow.entities import FileInfo
+from mlflow.exceptions import (ABORTED, DATA_LOSS, DEADLINE_EXCEEDED, ENDPOINT_NOT_FOUND, INTERNAL_ERROR, INVALID_STATE,
+                               NOT_FOUND, REQUEST_LIMIT_EXCEEDED, RESOURCE_DOES_NOT_EXIST, RESOURCE_EXHAUSTED,
+                               TEMPORARILY_UNAVAILABLE, ErrorCode, MlflowException)
+
+from composer.utils import dist
+from composer.utils.object_store.object_store import ObjectStore, ObjectStoreTransientError
+
+__all__ = ['MLFlowObjectStore']
+
+MLFLOW_DATABRICKS_TRACKING_URI = 'databricks'
+MLFLOW_DBFS_PATH_PREFIX = 'databricks/mlflow-tracking/'
+
+DEFAULT_MLFLOW_EXPERIMENT_NAME = 'mlflow-object-store'
+
+PLACEHOLDER_EXPERIMENT_ID = 'EXPERIMENT_ID'
+PLACEHOLDER_RUN_ID = 'RUN_ID'
```

Comment:
@dakinggg I'm not particularly attached to what the placeholders should be, but the original plan to use format strings such as `{mlflow_experiment_id}` and `{mlflow_run_id}` doesn't quite work because `str.format(...)` will throw a `KeyError` if not all format arguments are provided.